File: src/main/java/org/thundernetwork/minecraft/slashhub/SlashHub.java
package org.thundernetwork.minecraft.slashhub;

import com.google.inject.Inject;
import com.velocitypowered.api.event.proxy.ProxyInitializeEvent;
import com.velocitypowered.api.event.Subscribe;
import com.velocitypowered.api.plugin.Plugin;
import org.slf4j.Logger;

@Plugin(
        id = "slashhub",
        name = "Slash Hub",
        version = BuildConstants.VERSION
)
public class SlashHub {

    @Inject
    private Logger logger;

    @Subscribe
    public void onProxyInitialization(ProxyInitializeEvent event) {
    }
}

File: src/main/kotlin/org/thundernetwork/minecraft/slashhub/SlashHub.kt
package org.thundernetwork.minecraft.slashhub

import com.google.inject.Inject
import com.velocitypowered.api.event.Subscribe
import com.velocitypowered.api.event.proxy.ProxyInitializeEvent
import com.velocitypowered.api.plugin.Plugin
import org.slf4j.Logger

@Plugin(
    id = "slashhub",
    name = "Slash Hub",
    version = BuildConstants.VERSION,
    description = "Slash Hub",
    authors = ["ThunderNetworkRaD", "KillerBossOriginal"]
)
class SlashHub {
    @Inject
    private val logger: Logger? = null

    @Subscribe
    fun onProxyInitialization(event: ProxyInitializeEvent?) {
        if (logger === null) {
            throw NullPointerException("Logger is null")
        }
        logger.info("ThunderNetwork's SlashHub Loaded!")
    }
}
